### DISCUSSION TITLE
How to check memory left (stack and heap) for STM32F103C8 or STM32F103CB during runtime

### DISCUSSION THREAD

**Discussion by BRoboticsSHADO** [stm32duino/Arduino_Core_STM32#1472] [Q&A] [answered]
Hi

I would like to know how much memory left when running STM32F103C8 with arduino code, how shall i do that? The idea of checking is to check if there's any memory leak during runtime that cause by my code.

Best,
Samuel

**Comment by fpistm** [comment, marked as answer]
Hi @BRoboticsSHADO 
You can refers to this example:
https://github.com/stm32duino/STM32Examples/blob/master/examples/Benchmarking/MemoryAllocationStatistics/MemoryAllocationStatistics.ino

Hope this answer your question.
